cogs/archive.py

```diff
@@ -125,97 +174,110 @@ async def archive(self, ctx: "TeXBotApplicationContext", str_category_id: str) -
-                    category.name,
-                )
-                return
-
-        await ctx.respond("Category successfully archived", ephemeral=True)
+            # NOTE: Categories can not be placed inside other categories, so this will always be false, but is needed due to the typing of the method
+            if isinstance(channel, discord.CategoryChannel):
+                continue
+
+            await channel.edit(sync_permissions=True)
+
+        overwrite: discord.Member | discord.Role
+        for overwrite in category.overwrites:
+            await category.set_permissions(overwrite, overwrite=None)
+
+        if allow_archivist:
+            await category.set_permissions(
+                target=archivist_role,
+                read_messages=True,
+                read_message_history=True,
+            )
+
+        await category.edit(name=f"archive-{category.name}")
+
+        await initial_response.edit(
+            content=f":white_check_mark: Category '{category.name}' successfully archived."
+        )
+
+    @discord.slash_command(  # type: ignore[no-untyped-call, misc]
+        name="archive-channel",
+        description="Archives the selected channel.",
+    )
+    @discord.option(  # type: ignore[no-untyped-call, misc]
+        name="channel",
+        description="The channel to archive.",
+        input_type=str,
+        autocomplete=discord.utils.basic_autocomplete(autocomplete_get_non_archived_channels),  # type: ignore[arg-type]
+        required=True,
+        parameter_name="str_channel_id",
+    )
+    @discord.option(  # type: ignore[no-untyped-call, misc]
+        name="category",
+        description="The category to move the channel to.",
+        input_type=str,
+        autocomplete=discord.utils.basic_autocomplete(autocomplete_get_archival_categories),  # type: ignore[arg-type]
+        required=True,
+        parameter_name="str_category_id",
+    )
+    @CommandChecks.check_interaction_user_has_committee_role
+    @CommandChecks.check_interaction_user_in_main_guild
+    async def archive_channel(  # type: ignore[misc]
+        self, ctx: "TeXBotApplicationContext", str_channel_id: str, str_category_id: str
+    ) -> None:
+        """
+        Definition & callback response of the "archive-channel" command.
+
+        The "archive-channel" command moves the channel into the selected category
+        and syncs the permissions to the category's permissions.
+        """
+        # NOTE: Shortcut accessors are placed at the top of the function, so that the exceptions they raise are displayed before any further errors may be sent
+        main_guild: discord.Guild = self.bot.main_guild
+
+        if not re.fullmatch(r"\A\d{17,20}\Z", str_category_id) or not re.fullmatch(
+            r"\A\d{17,20}\Z", str_channel_id
+        ):
+            await self.command_send_error(
+                ctx=ctx, message=f"{str_category_id!r} is not a valid category ID."
+            )
+            return
```

Comment:
```suggestion
        if not re.fullmatch(r"\A\d{17,20}\Z", str_channel_id):
            await self.command_send_error(
                ctx=ctx, message=f"{str_channel_id!r} is not a valid channel ID."
            )
            return
```
Move regex validation of category ID below retrieval of channel (move to line 269).